modules/openapi-generator/src/main/resources/dart/libraries/dio/api.mustache

```diff
@@ -116,35 +119,15 @@ class {{classname}} {
       cancelToken: cancelToken,
       onSendProgress: onSendProgress,
       onReceiveProgress: onReceiveProgress,
-    );
-    {{#returnType}}
+    );{{^returnType}}
 
-    {{{returnType}}} _responseData;
-
-    try {
-{{#useBuiltValue}}{{>serialization/built_value/deserialize}}{{/useBuiltValue}}
-    } catch (error, stackTrace) {
-      throw DioError(
-        requestOptions: _response.requestOptions,
-        response: _response,
-        type: DioErrorType.other,
-        error: error,
-      )..stackTrace = stackTrace;
-    }
-
-    return Response<{{{returnType}}}>(
-      data: _responseData,
-      headers: _response.headers,
-      isRedirect: _response.isRedirect,
-      requestOptions: _response.requestOptions,
-      redirects: _response.redirects,
-      statusCode: _response.statusCode,
-      statusMessage: _response.statusMessage,
-      extra: _response.extra,
-    );{{/returnType}}{{^returnType}}
-    return _response;{{/returnType}}
+    return _response;{{/returnType}}{{#returnType}}{{#responses}}
+{{#isDefault}}{{>serialization/built_value/deserialize_response_type}}{{/isDefault}}{{#is2xx}}{{>serialization/built_value/deserialize_response_type}}{{/is2xx}}{{/responses}}{{/returnType}}
   }
 
   {{/operation}}
 }
+{{#inlineEnums}}
+{{>serialization/built_value/enum_inline}}
```

Comment:
These enums will cause collisions for some of my projects and likely for others as well.
I think enum handling for parameters should be a separate PR.

Reply:
Yes but there can be enums with the same name but different values for multiple operations in on API class.

Reply:
I think we can solve that by prepending the name of the api class to the name of the enum, what do you think ?

Reply:
Probably the name of the operation? That one might be long but I don't see another way.